core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -234,13 +234,19 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
         PositionData storage positionData = _getPositionData(msg.sender);
         require(positionData.requestPassTimestamp < getCurrentTime());
 
-        _removeCollateral(positionData.rawCollateral, positionData.withdrawalRequestAmount);
-        _removeCollateral(rawTotalPositionCollateral, positionData.withdrawalRequestAmount);
+        // If withdrawal request amount is > position collateral, then withdraw the full collateral amount.
+        FixedPoint.Unsigned memory amountToWithdraw = positionData.withdrawalRequestAmount;
+        if (positionData.withdrawalRequestAmount.isGreaterThan(_getCollateral(positionData.rawCollateral))) {
+            amountToWithdraw = _getCollateral(positionData.rawCollateral);
```

Comment:
this is great. in the worst case that your value has been eaten by fees below the value you have you get your maximum that you can withdraw.